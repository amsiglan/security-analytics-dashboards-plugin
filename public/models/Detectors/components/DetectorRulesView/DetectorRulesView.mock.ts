import { detectorMock, notificationsStart } from '../../../Interfaces.mock';

export default {
  detector: detectorMock,
  rulesCanFold: true,
  onEditClicked: jest.fn(),
  notifications: notificationsStart,
};
