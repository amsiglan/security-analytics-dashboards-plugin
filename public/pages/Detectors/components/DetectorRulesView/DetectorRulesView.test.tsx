/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from '@testing-library/react';
import props from '../../../../models/Detectors/components/DetectorRulesView/DetectorRulesView.mock';
import { expect } from '@jest/globals';
import { DetectorRulesView } from './DetectorRulesView';
jest.mock('../../../Rules/models/RulesViewModelActor.ts');

describe('<DetectorRulesView /> spec', () => {
  it('renders the component', () => {
    const view = render(<DetectorRulesView {...props} />);
    expect(view).toMatchSnapshot();
  });
});
