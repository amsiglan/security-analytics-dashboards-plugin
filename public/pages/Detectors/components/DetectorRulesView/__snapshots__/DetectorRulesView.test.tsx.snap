// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetectorRulesView /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="euiAccordion"
        title="View detection rules (2)"
      >
        <div
          class="euiAccordion__triggerWrapper"
        >
          <button
            aria-controls="detectorName"
            aria-expanded="false"
            class="euiAccordion__button"
            id="some_html_id"
            type="button"
          >
            <span
              class="euiAccordion__iconWrapper"
            >
              EuiIconMock
            </span>
            <span
              class="euiIEFlexWrapFix"
            >
              <div
                class="euiText euiText--medium"
              >
                <p>
                  View detection rules (2)
                </p>
              </div>
            </span>
          </button>
        </div>
        <div
          aria-labelledby="some_html_id"
          class="euiAccordion__childWrapper"
          id="detectorName"
          role="region"
          tabindex="-1"
        >
          <div>
            <div
              class=""
            >
              <div
                class="euiSpacer euiSpacer--l"
              />
              <div
                class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
              >
                <div>
                  <div
                    class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
                  >
                    <div
                      class="euiFlexItem euiSearchBar__searchHolder"
                    >
                      <div
                        class="euiFormControlLayout euiFormControlLayout--fullWidth"
                      >
                        <div
                          class="euiFormControlLayout__childrenWrapper"
                        >
                          <input
                            aria-label="This is a search bar. After typing your query, hit enter to filter the results lower in the page."
                            class="euiFieldSearch euiFieldSearch--fullWidth"
                            placeholder="Search rules"
                            type="search"
                            value=""
                          />
                          <div
                            class="euiFormControlLayoutIcons"
                          >
                            <span
                              class="euiFormControlLayoutCustomIcon"
                            >
                              EuiIconMock
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="euiFlexItem euiFlexItem--flexGrowZero euiSearchBar__filtersHolder"
                    >
                      <div
                        class="euiFilterGroup"
                      >
                        <div
                          class="euiPopover euiPopover--anchorDownCenter"
                          id="field_value_selection_0"
                        >
                          <div
                            class="euiPopover__anchor"
                          >
                            <button
                              class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon"
                              type="button"
                            >
                              <span
                                class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                              >
                                EuiIconMock
                                <span
                                  class="euiButtonEmpty__text"
                                >
                                  <span
                                    class="euiFilterButton__textShift"
                                    data-text="Rule Type"
                                    title="Rule Type"
                                  >
                                    Rule Type
                                  </span>
                                </span>
                              </span>
                            </button>
                          </div>
                        </div>
                        <div
                          class="euiPopover euiPopover--anchorDownCenter"
                          id="field_value_selection_1"
                        >
                          <div
                            class="euiPopover__anchor"
                          >
                            <button
                              class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon"
                              type="button"
                            >
                              <span
                                class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                              >
                                EuiIconMock
                                <span
                                  class="euiButtonEmpty__text"
                                >
                                  <span
                                    class="euiFilterButton__textShift"
                                    data-text="Rule Severity"
                                    title="Rule Severity"
                                  >
                                    Rule Severity
                                  </span>
                                </span>
                              </span>
                            </button>
                          </div>
                        </div>
                        <div
                          class="euiPopover euiPopover--anchorDownCenter"
                          id="field_value_selection_2"
                        >
                          <div
                            class="euiPopover__anchor"
                          >
                            <button
                              class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon"
                              type="button"
                            >
                              <span
                                class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                              >
                                EuiIconMock
                                <span
                                  class="euiButtonEmpty__text"
                                >
                                  <span
                                    class="euiFilterButton__textShift"
                                    data-text="Source"
                                    title="Source"
                                  >
                                    Source
                                  </span>
                                </span>
                              </span>
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="euiSpacer euiSpacer--l"
                  />
                  <div
                    class="euiBasicTable euiBasicTable-loading"
                  >
                    <div>
                      <div
                        class="euiTableHeaderMobile"
                      >
                        <div
                          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
                        >
                          <div
                            class="euiFlexItem euiFlexItem--flexGrowZero"
                          />
                          <div
                            class="euiFlexItem euiFlexItem--flexGrowZero"
                          >
                            <div
                              class="euiTableSortMobile"
                            >
                              <div
                                class="euiPopover euiPopover--anchorDownRight"
                              >
                                <div
                                  class="euiPopover__anchor"
                                >
                                  <button
                                    class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--flushRight"
                                    type="button"
                                  >
                                    <span
                                      class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                                    >
                                      EuiIconMock
                                      <span
                                        class="euiButtonEmpty__text"
                                      >
                                        Sorting
                                      </span>
                                    </span>
                                  </button>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <table
                        class="euiTable euiTable--responsive"
                        id="some_html_id"
                        tabindex="-1"
                      >
                        <caption
                          class="euiScreenReaderOnly euiTableCaption"
                        />
                        <thead>
                          <tr>
                            <th
                              aria-live="polite"
                              aria-sort="none"
                              class="euiTableHeaderCell"
                              data-test-subj="tableHeaderCell_title_0"
                              role="columnheader"
                              scope="col"
                              style="width: 30%;"
                            >
                              <button
                                class="euiTableHeaderButton"
                                data-test-subj="tableHeaderSortButton"
                                type="button"
                              >
                                <span
                                  class="euiTableCellContent"
                                >
                                  <span
                                    class="euiTableCellContent__text"
                                    title="Rule name"
                                  >
                                    Rule name
                                  </span>
                                </span>
                              </button>
                            </th>
                            <th
                              aria-live="polite"
                              aria-sort="none"
                              class="euiTableHeaderCell"
                              data-test-subj="tableHeaderCell_level_1"
                              role="columnheader"
                              scope="col"
                              style="width: 10%;"
                            >
                              <button
                                class="euiTableHeaderButton"
                                data-test-subj="tableHeaderSortButton"
                                type="button"
                              >
                                <span
                                  class="euiTableCellContent"
                                >
                                  <span
                                    class="euiTableCellContent__text"
                                    title="Rule Severity"
                                  >
                                    Rule Severity
                                  </span>
                                </span>
                              </button>
                            </th>
                            <th
                              aria-live="polite"
                              aria-sort="none"
                              class="euiTableHeaderCell"
                              data-test-subj="tableHeaderCell_category_2"
                              role="columnheader"
                              scope="col"
                              style="width: 10%;"
                            >
                              <button
                                class="euiTableHeaderButton"
                                data-test-subj="tableHeaderSortButton"
                                type="button"
                              >
                                <span
                                  class="euiTableCellContent"
                                >
                                  <span
                                    class="euiTableCellContent__text"
                                    title="Log type"
                                  >
                                    Log type
                                  </span>
                                </span>
                              </button>
                            </th>
                            <th
                              aria-live="polite"
                              aria-sort="none"
                              class="euiTableHeaderCell"
                              data-test-subj="tableHeaderCell_source_3"
                              role="columnheader"
                              scope="col"
                              style="width: 10%;"
                            >
                              <button
                                class="euiTableHeaderButton"
                                data-test-subj="tableHeaderSortButton"
                                type="button"
                              >
                                <span
                                  class="euiTableCellContent"
                                >
                                  <span
                                    class="euiTableCellContent__text"
                                    title="Source"
                                  >
                                    Source
                                  </span>
                                </span>
                              </button>
                            </th>
                            <th
                              class="euiTableHeaderCell"
                              data-test-subj="tableHeaderCell_description_4"
                              role="columnheader"
                              scope="col"
                            >
                              <span
                                class="euiTableCellContent"
                              >
                                <span
                                  class="euiTableCellContent__text"
                                  title="Description"
                                >
                                  Description
                                </span>
                              </span>
                            </th>
                          </tr>
                        </thead>
                        <tbody>
                          <tr
                            class="euiTableRow"
                          >
                            <td
                              class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
                              colspan="5"
                            >
                              <div
                                class="euiTableCellContent euiTableCellContent--alignCenter"
                              >
                                <span
                                  class="euiTableCellContent__text"
                                >
                                  No items found
                                </span>
                              </div>
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="euiAccordion"
      title="View detection rules (2)"
    >
      <div
        class="euiAccordion__triggerWrapper"
      >
        <button
          aria-controls="detectorName"
          aria-expanded="false"
          class="euiAccordion__button"
          id="some_html_id"
          type="button"
        >
          <span
            class="euiAccordion__iconWrapper"
          >
            EuiIconMock
          </span>
          <span
            class="euiIEFlexWrapFix"
          >
            <div
              class="euiText euiText--medium"
            >
              <p>
                View detection rules (2)
              </p>
            </div>
          </span>
        </button>
      </div>
      <div
        aria-labelledby="some_html_id"
        class="euiAccordion__childWrapper"
        id="detectorName"
        role="region"
        tabindex="-1"
      >
        <div>
          <div
            class=""
          >
            <div
              class="euiSpacer euiSpacer--l"
            />
            <div
              class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
            >
              <div>
                <div
                  class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
                >
                  <div
                    class="euiFlexItem euiSearchBar__searchHolder"
                  >
                    <div
                      class="euiFormControlLayout euiFormControlLayout--fullWidth"
                    >
                      <div
                        class="euiFormControlLayout__childrenWrapper"
                      >
                        <input
                          aria-label="This is a search bar. After typing your query, hit enter to filter the results lower in the page."
                          class="euiFieldSearch euiFieldSearch--fullWidth"
                          placeholder="Search rules"
                          type="search"
                          value=""
                        />
                        <div
                          class="euiFormControlLayoutIcons"
                        >
                          <span
                            class="euiFormControlLayoutCustomIcon"
                          >
                            EuiIconMock
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="euiFlexItem euiFlexItem--flexGrowZero euiSearchBar__filtersHolder"
                  >
                    <div
                      class="euiFilterGroup"
                    >
                      <div
                        class="euiPopover euiPopover--anchorDownCenter"
                        id="field_value_selection_0"
                      >
                        <div
                          class="euiPopover__anchor"
                        >
                          <button
                            class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon"
                            type="button"
                          >
                            <span
                              class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                            >
                              EuiIconMock
                              <span
                                class="euiButtonEmpty__text"
                              >
                                <span
                                  class="euiFilterButton__textShift"
                                  data-text="Rule Type"
                                  title="Rule Type"
                                >
                                  Rule Type
                                </span>
                              </span>
                            </span>
                          </button>
                        </div>
                      </div>
                      <div
                        class="euiPopover euiPopover--anchorDownCenter"
                        id="field_value_selection_1"
                      >
                        <div
                          class="euiPopover__anchor"
                        >
                          <button
                            class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon"
                            type="button"
                          >
                            <span
                              class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                            >
                              EuiIconMock
                              <span
                                class="euiButtonEmpty__text"
                              >
                                <span
                                  class="euiFilterButton__textShift"
                                  data-text="Rule Severity"
                                  title="Rule Severity"
                                >
                                  Rule Severity
                                </span>
                              </span>
                            </span>
                          </button>
                        </div>
                      </div>
                      <div
                        class="euiPopover euiPopover--anchorDownCenter"
                        id="field_value_selection_2"
                      >
                        <div
                          class="euiPopover__anchor"
                        >
                          <button
                            class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon"
                            type="button"
                          >
                            <span
                              class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                            >
                              EuiIconMock
                              <span
                                class="euiButtonEmpty__text"
                              >
                                <span
                                  class="euiFilterButton__textShift"
                                  data-text="Source"
                                  title="Source"
                                >
                                  Source
                                </span>
                              </span>
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="euiSpacer euiSpacer--l"
                />
                <div
                  class="euiBasicTable euiBasicTable-loading"
                >
                  <div>
                    <div
                      class="euiTableHeaderMobile"
                    >
                      <div
                        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
                      >
                        <div
                          class="euiFlexItem euiFlexItem--flexGrowZero"
                        />
                        <div
                          class="euiFlexItem euiFlexItem--flexGrowZero"
                        >
                          <div
                            class="euiTableSortMobile"
                          >
                            <div
                              class="euiPopover euiPopover--anchorDownRight"
                            >
                              <div
                                class="euiPopover__anchor"
                              >
                                <button
                                  class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--flushRight"
                                  type="button"
                                >
                                  <span
                                    class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                                  >
                                    EuiIconMock
                                    <span
                                      class="euiButtonEmpty__text"
                                    >
                                      Sorting
                                    </span>
                                  </span>
                                </button>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <table
                      class="euiTable euiTable--responsive"
                      id="some_html_id"
                      tabindex="-1"
                    >
                      <caption
                        class="euiScreenReaderOnly euiTableCaption"
                      />
                      <thead>
                        <tr>
                          <th
                            aria-live="polite"
                            aria-sort="none"
                            class="euiTableHeaderCell"
                            data-test-subj="tableHeaderCell_title_0"
                            role="columnheader"
                            scope="col"
                            style="width: 30%;"
                          >
                            <button
                              class="euiTableHeaderButton"
                              data-test-subj="tableHeaderSortButton"
                              type="button"
                            >
                              <span
                                class="euiTableCellContent"
                              >
                                <span
                                  class="euiTableCellContent__text"
                                  title="Rule name"
                                >
                                  Rule name
                                </span>
                              </span>
                            </button>
                          </th>
                          <th
                            aria-live="polite"
                            aria-sort="none"
                            class="euiTableHeaderCell"
                            data-test-subj="tableHeaderCell_level_1"
                            role="columnheader"
                            scope="col"
                            style="width: 10%;"
                          >
                            <button
                              class="euiTableHeaderButton"
                              data-test-subj="tableHeaderSortButton"
                              type="button"
                            >
                              <span
                                class="euiTableCellContent"
                              >
                                <span
                                  class="euiTableCellContent__text"
                                  title="Rule Severity"
                                >
                                  Rule Severity
                                </span>
                              </span>
                            </button>
                          </th>
                          <th
                            aria-live="polite"
                            aria-sort="none"
                            class="euiTableHeaderCell"
                            data-test-subj="tableHeaderCell_category_2"
                            role="columnheader"
                            scope="col"
                            style="width: 10%;"
                          >
                            <button
                              class="euiTableHeaderButton"
                              data-test-subj="tableHeaderSortButton"
                              type="button"
                            >
                              <span
                                class="euiTableCellContent"
                              >
                                <span
                                  class="euiTableCellContent__text"
                                  title="Log type"
                                >
                                  Log type
                                </span>
                              </span>
                            </button>
                          </th>
                          <th
                            aria-live="polite"
                            aria-sort="none"
                            class="euiTableHeaderCell"
                            data-test-subj="tableHeaderCell_source_3"
                            role="columnheader"
                            scope="col"
                            style="width: 10%;"
                          >
                            <button
                              class="euiTableHeaderButton"
                              data-test-subj="tableHeaderSortButton"
                              type="button"
                            >
                              <span
                                class="euiTableCellContent"
                              >
                                <span
                                  class="euiTableCellContent__text"
                                  title="Source"
                                >
                                  Source
                                </span>
                              </span>
                            </button>
                          </th>
                          <th
                            class="euiTableHeaderCell"
                            data-test-subj="tableHeaderCell_description_4"
                            role="columnheader"
                            scope="col"
                          >
                            <span
                              class="euiTableCellContent"
                            >
                              <span
                                class="euiTableCellContent__text"
                                title="Description"
                              >
                                Description
                              </span>
                            </span>
                          </th>
                        </tr>
                      </thead>
                      <tbody>
                        <tr
                          class="euiTableRow"
                        >
                          <td
                            class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
                            colspan="5"
                          >
                            <div
                              class="euiTableCellContent euiTableCellContent--alignCenter"
                            >
                              <span
                                class="euiTableCellContent__text"
                              >
                                No items found
                              </span>
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
